interfaces.go

```diff
@@ -67,8 +71,10 @@ type Permission interface {
 	SetPermissionProvider(PermissionProvider)
 }
 
+type PermissionProviderGetPermission func(permissionProvider PermissionProvider, role Role, permission string) Permission
```

Comment:
What if you could name this `Permissioner` maybe the names could be a wee bit shorter.
`SetPermissioner` vs `SetGetPermission`.  Just a thought.